tests/unit/test.errors.js

```diff
@@ -40,13 +40,28 @@ describe('errors', () => {
       }
     }
 
+    class ErrorWithErrno extends Error {
+      errno: string;
+      constructor() {
+        super('pretend this is a system error');
+        this.errno = 'SOME_CODE';
```

Comment:
this should be something like `this.errno = 53`, which means [ENOTEMPTY](https://github.com/rvagg/node-errno)